subsys/bluetooth/controller/ll_sw/ull_central_iso.c

```diff
@@ -578,34 +599,30 @@ uint8_t ll_cig_remove(uint8_t cig_id)
 		ll_remove_iso_path(cis->lll.handle, BT_HCI_DATAPATH_DIR_CTLR_TO_HOST);
 		ll_remove_iso_path(cis->lll.handle, BT_HCI_DATAPATH_DIR_HOST_TO_CTLR);
 
-		ll_conn_iso_stream_release(cis);
+		has_cis = true;
 	}
 
-	/* Clear configuration */
-	cig->cis_count = 0;
-
-	/* Release the CIG instance */
-	ll_conn_iso_group_release(cig);
+	if (has_cis) {
+		/* Clear configuration only - let CIS disconnection release instance */
+		cig->cis_count = 0;
 
-	return BT_HCI_ERR_SUCCESS;
-}
```

Comment:
The recent change here (line 567 to 591) got lost somehow. Please merge it in correctly.

Reply:
The changes in this PR are required according to the babblesim unicast_client test to pass. Left side, calling `ll_conn_iso_group_release()`  without terminating active CIS does not seem right.

Reply:
There can be no active CIS at this point, as you cannot remove the CIG if it has active CISes. This is guarded by `cig->started`. Regarding the unicast_client babblesim test, this code is from PR #53450, which fixes that test (Issue #53301).

Reply:
Hmm, it returns with `BT_HCI_ERR_INSUFFICIENT_RESOURCES`. Has this been rebased so that https://github.com/zephyrproject-rtos/zephyr/pull/53450 has been merged when running the test? 

The change in https://github.com/zephyrproject-rtos/zephyr/pull/54099/commits/0963e441d63ff438f475b85270eae6c6b7497542 works with upstream Zephyr.

Maybe it's the `CONFIG_BT_ISO_MAX_CHAN=2` change that breaks something? Could you try to set it back to the default (1)?

Reply:
> Has this been rebased so that https://github.com/zephyrproject-rtos/zephyr/pull/53450 has been merged when running the test?

As this PR enables `CONFIG_BT_CTLR_CENTRAL_ISO=y` and `CONFIG_BT_CTLR_PERIPHERAL_ISO=y`, the change in #53450 causes the error in https://github.com/zephyrproject-rtos/zephyr/pull/52842#discussion_r1088798092.

There is no failure in the test when the test is run in CI in this PR.

If am right, `CONFIG_BT_ISO_MAX_CHAN=2` is required by `CONFIG_BT_AUDIO_UNICAST_CLIENT_GROUP_STREAM_COUNT=2`

Reply:
OK, let's leave this for now. I will make a new PR when the dust settles, if necessary.

Reply:
> If am right, CONFIG_BT_ISO_MAX_CHAN=2 is required by CONFIG_BT_AUDIO_UNICAST_CLIENT_GROUP_STREAM_COUNT=2

You are correct. I wonder how I didn't see that issue when running the tests locally.